packages/sdk/src/browser-wallets/xverse/index.ts

```diff
@@ -1,278 +1,91 @@
-import * as ecc from "@bitcoinerlab/secp256k1";
-import { initEccLib, Psbt } from "bitcoinjs-lib";
-import {
-  AddressPurpose,
-  getAddress,
-  GetAddressOptions,
-  GetAddressResponse,
-  signMessage as xverseSignMessage,
-  SignMessageOptions as XverseSignMessageOptions,
-  SignMessageResponse as XverseSignMessageResponse,
-  signTransaction,
-  SignTransactionOptions,
-  SignTransactionResponse,
-} from "sats-connect";
+// @vitest-environment happy-dom
+import { Psbt } from "bitcoinjs-lib";
+import { BitcoinProvider } from "sats-connect";
 
-import { getAddressFormat } from "../../addresses";
-import type { BrowserWalletNetwork } from "../../config/types";
+import { BrowserWalletNetwork } from "../../config/types";
+import { BrowserWalletNotInstalledError, OrditSDKError } from "../../errors";
 import {
-  BrowserWalletExtractTxFromNonFinalizedPsbtError,
-  BrowserWalletNotInstalledError,
-  BrowserWalletRequestCancelledByUserError,
-  BrowserWalletSigningError,
-  OrditSDKError,
-} from "../../errors";
-import type { BrowserWalletSignResponse, WalletAddress } from "../types";
-import { NETWORK_TO_BITCOIN_NETWORK_TYPE } from "./constants";
-import type { XverseSignPSBTOptions } from "./types";
-import { fromXOnlyToFullPubkey } from "./utils";
-
-// Included here as browser-wallets is an individual entry-point.
-// Required for all signing operations to function within xverse since inputs need to be finalized.
-// This function will manage its own instance.
-initEccLib(ecc);
+  satsConnectWalletGetAddresses,
+  satsConnectWalletSignMessage,
+  satsConnectWalletSignPsbt,
+} from "../sats-connect";
+import type { SatsConnectSignPSBTOptions } from "../sats-connect/types";
+import { BrowserWalletSignResponse, WalletAddress } from "../types";
 
 /**
- * Checks if the browser wallet extension is installed.
+ * Checks if the Xverse extension is installed.
  *
  * @returns `true` if installed, `false` otherwise.
  * @throws {OrditSDKError} Function is called outside a browser without `window` object
  */
-function isInstalled() {
+function isInstalled(): boolean {
   if (typeof window === "undefined") {
     throw new OrditSDKError("Cannot call this function outside a browser");
   }
 
-  return typeof window.BitcoinProvider !== "undefined";
+  return typeof window.XverseProviders?.BitcoinProvider !== "undefined";
+}
+
+async function getXverseWalletProvider(): Promise<BitcoinProvider | undefined> {
```

Comment:
This should not be undefined given that `isInstalled` will throw an error.

We can make a non-null assertion here.

Reply:
Fixed.